### ISSUE TITLE
AutoTokenizer and AutoModel support

### ISSUE THREAD

**Issue by MikkoMMM** [cloveranon/Clover-Edition#112]
There are a lot of models here: https://huggingface.co/models
I tried the Finnish cased one and ran into problems. I tried to convert the config and vocab files into the same sort of format that the GPT models are in and had an empty merges.txt, but I only managed to get a long line of random Finnish word parts for my reward.

Then I noticed that the entries on that page have instructions on "how to use the model directly in the transformers library". A lot, maybe all, of them use the AutoTokenizer and the AutoModel classes instead of the GPT2Tokenizer and GPT2LMHeadModel classes gpt2generator.py is currently using. I tried switching those names and deleted any code that produced errors after that, and then it at least ran though produced a long string of mostly one word. But I think it served as a proof of concept that it might not be beyond the realms of possibility to use those models listed on that page as-is if we can figure out the needed modifications to gpt2generator.py. I have no idea what those are.

**Comment by devfag** [comment]
I agree that it would be very useful to have access to these models in our version. We would need to rewrite a lot of our regex/action detection to support the different language models, but we could possibly leverage the help of the international boards to get the proper grammars we need to be parsing.

**Comment by cloveranon** [comment]
Try getting those models to generate text using just the transformers library. See if they generate text worth reading. So far as I know GPT2 is miles better at text generation than anything else right now. They allegedly spent tens of thousands of dollars to train it on special hardware. And with a very large private dataset. When a new good model is published it might be worth the effort to switch.

**Comment by Rioghasarig** [comment]
Huggingface has example code in which they demonstrate language modeling with several different models. If we follow the way they do it, it shouldn't be that hard to use any model from the huggingface library.
